### DISCUSSION TITLE
required function is not working after overwritten the onChange function

### DISCUSSION THREAD

**Discussion by yip-anthony-principal** [react-hook-form/react-hook-form#6532] [Q&A]
The server's data structure for the agreement is `Boolean`. Everything works fine after transform the format in `onChange`. However, the form became unsubmittable for the false value field. The codesandbox example can provide a better picture for this issue.

https://codesandbox.io/s/react-hook-form-v7-controller-forked-cork2?file=/src/Mui.js

I think the problem is somewhat related to the `onChange`'s `false` triggered `required` to the `false` and then cause `error` in `formState`.

**Comment by bluebill1049** [comment]
https://react-hook-form.com/api/useform/register

<img width="1353" alt="Screen Shot 2021-09-13 at 8 35 58 pm" src="https://user-images.githubusercontent.com/10513364/133069351-5cdc88e2-597a-4b5b-8017-d848a4ef68e5.png">


https://github.com/react-hook-form/react-hook-form/pull/5620

this PR make it easeir
